### ISSUE TITLE
Namespace clash between control name and input

### ISSUE THREAD

**Issue by lhasadreams** [inspec/inspec#4936]
InSpec V4.16.0 and before execute a control without error
Raised on behalf of Paul Burnet (Lloyds Banking Group)

# Describe the problem
This control use to work fine up to InSpec v4.16.0, using the latest InSpec (v4.18.97) gives an error.
```
control 'my_control' do
only_if {input('my_control', value: 'false') == 'false' }
 <ommitted>
end
```

When executing the control InSpec gives the following error:
```
 ×  Control Source Code Error ./controls/IAM/iam_benchmark_1_57_27_58_cloudsql_mysql_restrict_stop_delete.rb:7
     string not matched
```

This can be traced to a namespace error between the control name and the inout name - this a regression.
Changing the control to:
```
control 'my_control' do
only_if {input('skip_my_control', value: 'false') == 'false' }
 <ommitted>
end
```
Removes the error.

The latest version of InSpec should execute the above control without error. The customer doesn't really want to change the code to work around the regression.

## Possible Solution
(1) Fix the code regression, @clintoncwolfe and @TheLunaticScripter indicated that it may be a side effect of the Waiver work.
(2) Pin to the earlier version of InSpec - not an option as later functionality required.
(3) Change all the input names - customer would prefer not to.
(3) Change completely and use Waivers instead - I am testing this with the customer, however Waivers are beta at the moment and InSpec is being used in a security critical area.

**Comment by Schwad** [comment]
I can confirm that I've run the tests/output laid out above and replicated the error. 

```
control 'my_control' do
only_if {input('my_control', value: 'false') == 'false' }
 <ommitted>
end
```

Fails current inspec, passes 4.16.0. Will try and find the breaking commit 👍

**Comment by Schwad** [comment]
Having reviewed our history here, confirmed that the commit that causes the above syntax to fail is: https://github.com/inspec/inspec/commit/43bc7e034387eccaff1b889c02dc7c57aae33909

To reproduce, cd into local copy of repo and run:

```
git checkout 43bc7e034387eccaff1b889c02dc7c57aae33909
bundle install
bundle exec inspec-bin/bin/inspec exec path/to/profile/with/above/control -t <my_target>
#=> ×  Control Source Code Error schwad_sandbox/controls/example.rb:22 
     undefined method `key?' for "false":String
```

Then go to prior commit

```
git checkout aab45b507fe625d81a2104f6f9b9362471e566ed
rm Gemfile.lock
bundle install
bundle exec inspec-bin/bin/inspec exec path/to/profile/with/above/control -t <my_target>
#=> ✔  Passes
```

**Comment by TheLunaticScripter** [comment]
This is part of the waivers feature and I don't think this is a bug. I think that the input needs to be renamed to be something other than the control name. Based on the code example shown above this is exactly what waivers does by specifying a control id in a waiver file it will skip the control. I think the recommendation should be to use the new waivers functionality.

**Comment by stevendanna** [comment]
We broke a profile that was not using any internal features in a non-major release.  While we can decide that we don't want to fix that, let's avoid calling this anything but a bug.

**Comment by TheLunaticScripter** [comment]
Fair this was a breaking change. Let's be sure to get all the information around that commit before making a decision on what we do. I care less about what happened and more that waivers is an active feature and breaking that also has ramifications.

**Comment by kekaichinose** [comment]
@lhasadreams per Possible Solution 3 above, how's the testing going with the customer with using waivers? It became official in the last two weeks so would like to see how it solves the customer's problem. Thanks!

**Comment by Schwad** [comment]
I had a look at our existing codebase, using the example from @lhasadreams to see what could be done. 

The PR above is the result of that work. It preserves existing functionality & waivers, but does allow the user to have a clash between the input and control namespace without breaking. Tested locally as well.  

Just working to pass CI 😅

Thanks!

**Comment by Schwad** [comment]
@lhasadreams I am pleased to report that this issue has been fixed in InSpec 4.18.111, which we released about half an hour ago.
